src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonPropertiesView.ts

```diff
@@ -148,6 +148,24 @@ export class ExecutionPlanComparisonPropertiesView extends ExecutionPlanProperti
 		this.populateTable(columns, tableRows);
 	}
 
+	public expandOrCollapsePropertiesTable(expand: boolean): void {
+		const columns: Slick.Column<Slick.SlickData>[] = this.getPropertyTableColumns();
+
+		let primaryProps = [];
+		let secondaryProps = [];
+		if (this._model.primaryElement?.properties) {
+			primaryProps = this._model.primaryElement.properties;
+		}
+		if (this._model.secondaryElement?.properties) {
+			secondaryProps = this._model.secondaryElement.properties;
+		}
+
+		let tableRows = this.convertPropertiesToTableRows(primaryProps, secondaryProps);
```

Comment:
Why do we have to do all this conversion stuff just to modify the expanded state? We should be able to access the rows directly and set them expanded collapsed there instead of creating a whole new set of data and redrawing it every time - look at exposing functionality for that in `treeGrid.ts` (we already have `setRowExpandedState` which is doing something similar)

Reply:
That was to trigger the UI refresh. For some reason, changing just the expanded property value to true or false for each collapsible row wasn't updating the UI. I noticed alphabetical sorting did something similar, which is why I did it for the collapse and expand functionality.

Reply:
So that's because these items are just raw data. You still need to refresh the grid (see `setRowExpandedState` where it's doing that). But the part you shouldn't need to do is recreate the whole list of objects every time you change this. The data is all the same - it's just that the expanded state is changing. 

So you should be able to just modify treeGrid to allow callers to set the expanded state of the items - then you'd just call something like `grid.setAllRowsExpandedState` or something which would loop through and update the expanded state for each item and then call rerenderGrid afterwards.